ln-gateway/src/ln.rs

```diff
@@ -66,3 +68,53 @@ impl LnRpc for Mutex<cln_rpc::ClnRpc> {
         }
     }
 }
+
+/// Control the behavoir of the LnRpc by exposing the same interface but attach configurations and
+/// methods to it.
+pub struct LnRpcAdapter {
```

Comment:
Just occurred to me that the adapter is a purely test concept. Can we define and implement it somewhere else closer to the test framework?

Reply:
mhm yes, my thought was to put it into the lightning module crate but maybe it should be in the integration test `fake.rs` ?
lets discuss on discord, maybe it makes sense to have a adapter in a non testing setting ?